api/envoy/api/v2/route/route_components.proto

```diff
@@ -696,6 +696,27 @@ message RouteAction {
     google.protobuf.BoolValue enabled = 2;
   }
 
+  // Describe how to match the path and rewrite it using a regular expression
+  // and a substitution string.
+  message RegexRewrite {
+    // The regular expression used to find portions of the path (excluding
+    // query parameters) that should be replaced. All matches in the path are
+    // replaced by the substitution string, though anchors in the regular
+    // expression can be used to replace just one occurrence of a pattern.
+    // Capture groups can be used in the pattern to extract portions of the
+    // path, and referenced in the substitution string.
+    type.matcher.RegexMatcher pattern = 1;
+
+    // The string that should be substituted into matching portions of the path.
+    // Capture groups in the pattern can be referenced in this string  Beware,
+    // however, that the syntax for referring to capture groups might vary
```

Comment:
nit: period before "Beware" and I think we should say that the "... syntax for referring to capture groups is defined by the regular expression engines."

Reply:
Sorry about the missing period.  The rewording is changed too in the latest commit.